basicclient.go

```diff
@@ -0,0 +1,251 @@
+}
+
+// PsaCipherEncrypt carries out symmetric encryption on plaintext using defined key/algorithm, returning ciphertext
+func (c BasicClient) PsaCipherEncrypt(keyName string, alg *algorithm.Cipher, plaintext []byte) ([]byte, error) {
+	opalg, err := algCipherAlgToWire(alg)
+	if err != nil {
+		return nil, err
+	}
+	return c.opclient.PsaCipherEncrypt(keyName, opalg, plaintext)
+}
+
+// PsaCipherDecrypt decrypts symmetrically encrypted ciphertext using defined key/algorithm, returning plaintext
+func (c BasicClient) PsaCipherDecrypt(keyName string, alg *algorithm.Cipher, ciphertext []byte) ([]byte, error) {
+	opalg, err := algCipherAlgToWire(alg)
+	if err != nil {
+		return nil, err
+	}
+	return c.opclient.PsaCipherDecrypt(keyName, opalg, ciphertext)
+}
+
+func (c BasicClient) PsaAeadDecrypt(keyName string, alg *algorithm.AeadAlgorithm, nonce, additionalData, ciphertext []byte) ([]byte, error) {
+	opalg, err := algAeadAlgToWire(alg)
+	if err != nil {
+		return nil, err
+	}
+	return c.opclient.PsaAeadDecrypt(keyName, opalg, nonce, additionalData, ciphertext)
+}
+
+func (c BasicClient) PsaAeadEncrypt(keyName string, alg *algorithm.AeadAlgorithm, nonce, additionalData, plaintext []byte) ([]byte, error) {
+	opalg, err := algAeadAlgToWire(alg)
+	if err != nil {
+		return nil, err
+	}
+	return c.opclient.PsaAeadEncrypt(keyName, opalg, nonce, additionalData, plaintext)
+}
+
+func (c BasicClient) PsaExportKey(keyName string) ([]byte, error) {
+	return c.opclient.PsaExportKey(keyName)
+}
+
+func (c BasicClient) PsaImportKey(keyName string, attributes *KeyAttributes, data []byte) error {
+	opattrs, err := attributes.toWireInterface()
+	if err != nil {
+		return err
+	}
+	return c.opclient.PsaImportKey(keyName, opattrs, data)
+}
+
+func (c BasicClient) PsaExportPublicKey(keyName string) ([]byte, error) {
+	return c.opclient.PsaExportPublicKey(keyName)
+}
+
+func (c BasicClient) PsaGenerateRandom(size uint64) ([]byte, error) {
+	return c.opclient.PsaGenerateRandom(size)
+}
+
+func (c BasicClient) PsaMACCompute(keyName string, alg *algorithm.MacAlgorithm, input []byte) ([]byte, error) {
+	opalg, err := algMacAlgToWire(alg)
+	if err != nil {
+		return nil, err
+	}
+	return c.opclient.PsaMACCompute(keyName, opalg, input)
+}
+
+func (c BasicClient) PsaMACVerify(keyName string, alg *algorithm.MacAlgorithm, input, mac []byte) error {
+	opalg, err := algMacAlgToWire(alg)
+	if err != nil {
+		return err
+	}
+	return c.opclient.PsaMACVerify(keyName, opalg, input, mac)
+}
+
+func (c BasicClient) PsaRawKeyAgreement(alg *algorithm.KeyAgreementRaw, privateKey string, peerKey []byte) ([]byte, error) {
+	opalg, err := algKeyAgreementRawAlgToWire(alg)
+	if err != nil {
+		return nil, err
+	}
+	return c.opclient.PsaRawKeyAgreement(opalg.GetRaw().Enum(), privateKey, peerKey)
+}
+
+func (c BasicClient) PsaAssymetricDecrypt(keyName string, alg *algorithm.AsymmetricEncryptionAlgorithm, salt, ciphertext []byte) ([]byte, error) {
```

Comment:
```suggestion
func (c BasicClient) PsaAsymetricDecrypt(keyName string, alg *algorithm.AsymmetricEncryptionAlgorithm, salt, ciphertext []byte) ([]byte, error) {
```

I think there are a few places where a similar typo was made